os/hal/ports/SN32/SN32F240B/sn32_isr.c

```diff
@@ -0,0 +1,126 @@
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+/**
+ * @file    SN32F24xx/sn32_isr.h
+ * @brief   SN32F24xx ISR handler code.
+ *
+ * @{
+ */
+
+#include "hal.h"
+
+/*===========================================================================*/
+/* Driver local definitions.                                                 */
+/*===========================================================================*/
+
+/*===========================================================================*/
+/* Driver exported variables.                                                */
+/*===========================================================================*/
+
+/*===========================================================================*/
+/* Driver local variables.                                                   */
+/*===========================================================================*/
+
+/*===========================================================================*/
+/* Driver local functions.                                                   */
+/*===========================================================================*/
+
+/*===========================================================================*/
+/* Driver interrupt handlers.                                                */
+/*===========================================================================*/
+
+
+//#if HAL_USE_GPT || HAL_USE_PWM || defined(__DOXYGEN__)
+///**
+// * @brief   CT16B0 interrupt handler.
+// *
+// * @isr
+// */
+//OSAL_IRQ_HANDLER(Vector7C) {
+//
+//  OSAL_IRQ_PROLOGUE();
+//
+//#if HAL_USE_GPT
+//#if SN32_GPT_USE_CT16B0
+//  gpt_lld_serve_interrupt(&GPTD1);
+//#endif
+//#endif
+//  OSAL_IRQ_EPILOGUE();
+//}
+//
+///**
+// * @brief   CT16B1 interrupt handler.
+// *
+// * @isr
+// */
+//OSAL_IRQ_HANDLER(Vector80) {
+//
+//  OSAL_IRQ_PROLOGUE();
+//
+//#if HAL_USE_GPT
+//#if SN32_GPT_USE_CT16B1
+//  gpt_lld_serve_interrupt(&GPTD2);
+//#endif
+//#endif
+//#if HAL_USE_PWM
+//#if SN32_PWM_USE_CT16B1
+//  pwm_lld_serve_interrupt(&PWMD1);
+//#endif
+//#endif
+//
+//  OSAL_IRQ_EPILOGUE();
+//}
+//#endif /* HAL_USE_GPT || HAL_USE_PWM */
+/*===========================================================================*/
+/* Driver exported functions.                                                */
+/*===========================================================================*/
+
+/**
+ * @brief   Enables IRQ sources.
+ *
+ * @notapi
+ */
+void irqInit(void) {

```

Comment:
What is the usecase of irqInit() and irqDeinit(), and the files sn32_isr.c / .h?

Shoudn't nvicEnableVector() and nvicDisableVector() be called by the gpt and ct drivers?

What are the SN32_CT16B0_SUPPRESS_ISR defines?

Reply:
These were leftovers from the awfully generic stm32 hal I traced for this. I am still tracing for a use case scenario, but it looks like I will probably amend and remove the isr.c/h completely